test/collection_test.js

```diff
@@ -167,23 +162,27 @@ describe("Collection", () => {
     });
 
     it("should create the expected request", () => {
-      sandbox.stub(requests, "createRecord");
+      sandbox.stub(requests, "createRequest");
 
       coll.createRecord(record);
 
-      sinon.assert.calledWith(requests.createRecord, "posts", record, {
-        bucket: "blog",
-        headers: {Foo: "Bar", Baz: "Qux"},
-      });
+      sinon.assert.calledWith(requests.createRequest, "/buckets/blog/collections/posts/records", {
+        data: record,
+        permissions: undefined
+      }, { headers: {Foo: "Bar", Baz: "Qux"} });
     });
 
     it("should accept a safe option", () => {
-      sandbox.stub(requests, "createRecord");
+      sandbox.stub(requests, "createRequest");
 
       coll.createRecord(record, {safe: true});
 
-      sinon.assert.calledWithMatch(requests.createRecord, "posts", record, {
-        safe: true
+      sinon.assert.calledWithMatch(requests.createRequest, "/buckets/blog/collections/posts/records", {
```

Comment:
To be noted we're heavily relying on `calledWithMatch` while I'm pretty sure most of our assertions should just pass using `calledWithExactly`.